### DISCUSSION TITLE
File listing - sort order

### DISCUSSION THREAD

**Discussion by szylllc** [community/community#31369] [Other Feature Feedback, Questions, & Ideas] [answered]
Is it possible to sort by file name or date ?
Right now the sort order appears to be random.

**Comment by look** [comment]
@szylllc do you mean sorting the results in code search?

**Reply by szylllc** [reply]
No, I'm just on the Main page of my repository. It's just a listing of all of my modules.

**Comment by airtower-luna** [comment]
On a repository page directories are show first, then files, each group sorted alphabetically. Do you possibly mean the list of your repositories? If yes, there's a "Sort" menu:

![grafik](https://user-images.githubusercontent.com/5814919/188179228-7665a4eb-0fcd-46cb-808e-bbccef0b6743.png)

The default is "last updated", so you see repositories with recent commits first.

**Comment by phillipakhn** [comment]
There is a sort button on your repo page that the top right.

**Comment by szylllc** [comment]
![GitHub](https://user-images.githubusercontent.com/110911952/188273452-2e4cee9a-3ef0-47e8-ab36-39b8d848658c.png)
Guys - the above is the page I am talking about. I see no sort option anywhere.

**Comment by airtower-luna** [comment, marked as answer]
That's a listing of files in your repository. The order is alphabetical, there's no option to change that.

**Reply by szylllc** [reply]
How can I get a View that supports the sorting of the files ?

**Comment by jfullstackdev** [comment]
what I can think of, if your repo is not fixed to folder structure, is to simply sort the related files by folder, that way, you can organize your files by folders, although folders will still be arranged alphabetically but you have your sorted files within each folder

**Reply by szylllc** [reply]
Hmmmm..I have no folders. re: " but you have your sorted files within each folder"
What sort order will they be ? By name ? No option to sort by date ?

**Reply by airtower-luna** [reply]
> No option to sort by date ?

Not in the GitHub web UI.

If you clone the repository to your machine you can use whatever tools you like to list the files. Note that the file modification date (which many file browsers can sort by) will usually not be the same as the date of the last commit that modified the file.

**Reply by szylllc** [reply]
re: "Not in the GitHub web UI."
Where can I post a request for an enhancement to provide this ?

BTW: I do not want to clone the repository to my local hard drive.

**Reply by airtower-luna** [reply]
The feedback category here is for feature requests, too.
